lstchain/reco/reconstructorCC.py

```diff
@@ -212,18 +210,19 @@ def template_interpolation(gain, times, t0, dt, a_hg, a_lg, size):
 
     """
     n, m = times.shape
+    size = a_hg.shape[0]
     out = np.empty((n, m))
     for i in range(n):
         for j in range(m):
             # Find the index before the requested time
             a = (times[i, j]-t0)/dt
             t = int(a)
-            if a < size:
+            if t+1 < size:
```

Comment:
why was this modified in one unit?

Reply:
Use <= instead of the awkward +  1 <

Reply:
Sorry I was wrong in what I was saying before since a<size also guaranties that t is less than size. The change is now quite old so I didn't fully remember. 
But we use both the t and t+1 element in the following code. So the error occurred when t+1 was equal to size. Since the array for interpolation was usually binned in fractions of nano second it was a rare bug I saw quite late. (there is no a==size issue).

I made the decision on checking on the value of t instead of a in the fix because this is what is used in the array afterwards.

The needed criteria is thus that t+1 is the index of an array element (so < size = len(array)).

@maxnoe <= wouldn't work, it goes in the wrong direction. It would be +2 <=

Reply:
ah, right, sorry...